pants-plugins/src/python/internal_backend/sitegen/tasks/sitegen.py

```diff
@@ -251,22 +251,46 @@ def generate_site_toc(config, precomputed, here):
 
   def recurse(tree, depth_so_far):
     for node in tree:
-      if 'heading' in node:
+      if 'heading' in node and 'pages' in node:
```

Comment:
I think the json would be clearer if collapsible sections had a different heading key word. Maybe `'collapsible_heading'`?